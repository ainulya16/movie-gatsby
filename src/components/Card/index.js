import React from 'react';
import styled from 'styled-components';
import {
  string,
  shape,
  objectOf,
  any,
  oneOfType,
  node,
  arrayOf,
} from 'prop-types';
import Icon from '@aia-digital/ui-library';

const CardBase = styled.div`
  @media (max-width: 992px) {
    padding: 0 40px;
  }
`;
const H6 = styled.h6`
  font-family: 'AIABody';
  font-weight: bold;
  max-lines: 2;
  font-size: 14pt;
  height: 30pt;
  margin-top: 20px;
  margin-bottom: 30px;
`;

const P = styled.p`
  font-family: 'AIABody';
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 5;
  -webkit-box-orient: vertical;
`;

const Card = ({ title, description, icon, children, ...attr }) => {
  return (
    <CardBase className="text-center" {...attr}>
      {/* {icon && <Icon name="call" />} */}
      {title && <H6>{title}</H6>}
      {description && <P>{description}</P>}
      {children}
    </CardBase>
  );
};

Card.propTypes = {
  title: string,
  icon: shape({ name: string, size: string, style: objectOf(any) }),
  description: string,
  children: oneOfType([node, arrayOf(node)]),
};
Card.defaultProps = {
  title: '',
  description: '',
  icon: null,
  children: null,
};

export default Card;
