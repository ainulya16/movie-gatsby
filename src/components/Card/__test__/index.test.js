/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Card from '..';

jest.mock('../../../static/img/icons.svg', () => 'icons.svg');

let component;
let wrapper;
describe('Card components', () => {
  beforeAll(() => {
    component = (
      <Card>
        <h1>test</h1>
        <div>
          <p>description</p>
        </div>
      </Card>
    );
    wrapper = shallow(component);
  });
  it('should be exist', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should be render children', () => {
    expect(wrapper.find('p')).toHaveLength(1);
  });
  it('should be with icon', () => {
    wrapper = shallow(
      <Card icon={{ name: 'menu-nav' }}>
        <p>test</p>
      </Card>,
    );
    expect(wrapper.find('Icon')).toHaveLength(1);
  });
  it('should match the snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
