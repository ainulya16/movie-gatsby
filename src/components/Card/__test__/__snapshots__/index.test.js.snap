// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card components should match the snapshot 1`] = `
@media (max-width:992px) {
  .c0 {
    padding: 0 40px;
  }
}

<div
  className="text-center c0"
>
  
  
  <h1>
    test
  </h1>
  <div>
    <p>
      description
    </p>
  </div>
</div>
`;
